parsl/usage_tracking/usage.py

```diff
@@ -98,90 +99,121 @@ def __init__(self, dfk, port=50077,
         self.domain_name = domain_name
         # The sock timeout will only apply to UDP send and not domain resolution
         self.sock_timeout = 5
-        self.UDP_PORT = port
+        self.udp_port = port
         self.procs = []
         self.dfk = dfk
         self.config = self.dfk.config
-        self.uuid = str(uuid.uuid4())
+        self.correlator_uuid = str(uuid.uuid4())
         self.parsl_version = PARSL_VERSION
-        self.python_version = "{}.{}.{}".format(sys.version_info.major,
-                                                sys.version_info.minor,
-                                                sys.version_info.micro)
-        self.tracking_enabled = self.check_tracking_enabled()
-        logger.debug("Tracking status: {}".format(self.tracking_enabled))
+        self.python_version = (
+            f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
+        )
+        self.tracking_level = self.check_tracking_level()
+        self.start_time = None
+        logger.debug(f"Tracking level: {self.tracking_level}")
+
+    def check_tracking_level(self):
+        """Check if tracking is enabled and return level.
+
+        Returns: int
+            - 0 : Tracking is disabled
+            - 1 : Tracking is enabled with level 1
+                  Share info about Parsl version, Python version, platform
+            - 2 : Tracking is enabled with level 2
+                  Share info about config + level 1
+            - 3 : Tracking is enabled with level 3
+                  Share info about app count, app fails, execution time + level 2
+        """
 
-    def check_tracking_enabled(self):
-        """Check if tracking is enabled.
+        level = 0
 
-        Tracking will be enabled unless either of these is true:
+        if self.config.usage_tracking:
+            level = 1 if self.config.usage_tracking is True else int(self.config.usage_tracking)
 
-            1. dfk.config.usage_tracking is set to False
-            2. Environment variable PARSL_TRACKING is set to false (case insensitive)
+        envvar = str(os.environ.get("PARSL_TRACKING", False)).lower()
+        if envvar in {"true", "1", "2", "3"}:
+            level = 1 if envvar == "true" else int(envvar)
```

Comment:
what happens if I set the envvar to something else? (such as Yes or On or Disable)?

Reply:
this is still an open question

Reply:
Anything other than 'True', 'true', '1', '2' or '3' will result in usage_tracking level 0